src/NServiceBus.Core/EndpointCreator.cs

```diff
@@ -91,9 +89,13 @@ void Initialize()
 
             recoverabilityComponent = new RecoverabilityComponent(settings);
 
+            featureComponent = new FeatureComponent(settings);
```

Comment:
So, if I'm understanding the behavior change you mentioned correctly, that means that feature information won't be in settings at all until this point.

If so, then this is absolutely an observable breaking change. `InitializeTransportComponent` calls `TransportDefinition.Initialize`, and transports definitely inspect feature information in there, for a variety of reasons.

Example:
https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.SqlServer/SqlServerTransportInfrastructure.cs#L25

Reply:
No that isn't correct, we just moved the point where we store the lambdas, we still invoke them at the same point

https://github.com/Particular/NServiceBus/blob/439d02fe779cd3ad8184d9d87a0d1341427e4a33/src/NServiceBus.Core/Features/FeatureActivator.cs#L44

(tests would have failed if that wasn't the case)

Reply:
Talked to @bording yesterday he's right, [the hacks we do in the transports to figure out if the timeout manager is disabled our not](https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.SqlServer/SqlServerTransportInfrastructure.cs#L25) assumes that the features "EnabledByDefault" state are present in the settings even though settings are not locked yet so the value can change. 

I will add a test to master that documents this behaviour and also raise an issue to find a better way to deal with this in the next major. Once that is done I'll move the defaults setting part of the features back to the same location in the code to make the test pass for this change as well

Reply:
Nice find. Wasn't aware of those hacks

Reply:
https://github.com/Particular/NServiceBus/pull/5429 adds a test to document this.

I will update this PR once its merged